src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -748,9 +757,19 @@ export class ResultsViewPageStore {
         }
     });
 
-    readonly coexpressionTabMolecularProfiles = remoteData<MolecularProfile[]>({
+    readonly coexpressionTabMolecularProfilesXGene = remoteData<MolecularProfile[]>({
+        await:()=>[this.molecularProfilesWithData],
+        invoke:()=>Promise.resolve(filterAndSortProfilesX("gene", this.molecularProfilesWithData.result!))
```

Comment:
why do you have a separate function for filter and sort x axis profiles vs y profiles?